op-challenger/fault/agent.go

```diff
@@ -45,12 +49,23 @@ func (a *Agent) PerformActions() {
 
 // move determines & executes the next move given a claim pair
 func (a *Agent) move(claim, parent Claim) error {
-	move, err := a.solver.NextMove(claim)
-	if err != nil || move == nil {
+	nextMove, err := a.solver.NextMove(claim)
+	if err != nil {
+		a.log.Warn("Failed to execute the next move", "err", err)
```

Comment:
I think it's acceptable insofar as the error log explicitly points to where that error was produced. In this case, it would be nice if the logger would prefix this warning with something like `op-challenger::fault-agent`.

Reply:
> bit sceptical of both logging and returning an error

Yea, figuring out where to log is always a little tricky. We don't have logging at a higher level than this right now so if we don't log, the error gets dropped, but I should revisit it when I get to error handling

> would be nice if the logger would prefix this warning with something like op-challenger::fault-agent

No need. the `op-challenger` component is implied via the source of the logs & the fault game isn't necessary yet, but if we do want it the place to add it is to the logger that is passed into the game. So when creating the agent logger, it'd get `log.New("game", "fault")` and then every log would have that metadata.